org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/binding/BindingsView.java

```diff
@@ -252,7 +260,42 @@ public void connectionRemoved(IConnectedEngineFacade server) {
     for (var server : SonarLintCorePlugin.getServersManager().getServers()) {
       server.addConnectedEngineListener(serverListener);
     }
-
+    
+    // add listener for when project is opened / closed / deleted
+    projectListener = event -> {
+      if (event.getBuildKind() == IResourceChangeEvent.PRE_CLOSE
+        || event.getBuildKind() == IResourceChangeEvent.PRE_DELETE) {
+        refreshView();
+        return;
+      }
+      
+      if (event.getDelta() == null) {
+        return;
+      }
+      
+      try {
+        event.getDelta().accept((IResourceDelta delta) -> {
+          var rsrc = delta.getResource();
+          if (((rsrc.getType() & IResource.PROJECT) != 0) 
+            && rsrc.getProject().isOpen()
+            && delta.getKind() == IResourceDelta.CHANGED
+            && ((delta.getFlags() & IResourceDelta.OPEN) != 0)) {
+            refreshView();
```

Comment:
Shouldn't you return "false" after that, to prevent multiple calls to "refresh". In fact, it might be more robust to collect a boolean "shouldRefresh", and only call "refreshView" outside the delta visitor.

Reply:
Done